### PR TITLE
New header bar and d2 dependency upgrades

### PR DESCRIPTION
**Pull request by turban**
This PR includes:  
- The new standard DHIS2 header bar
- Upgrade of all d2 and d2-ui dependencies
- The old MUI theme provider is removed

There is a z-index issue with the header bar that needs to be fixed before merging this PR into master. @adeelshahid

### CODE REVIEW

File: src/store/index.js [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
 
+const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+
 const store = createStore(
     reducer,
-    process.env.NODE_ENV === 'development' &&
-        window.__REDUX_DEVTOOLS_EXTENSION__ &&
-        window.__REDUX_DEVTOOLS_EXTENSION__(),
-    applyMiddleware(thunk, epicMiddleware)
+    composeEnhancer(applyMiddleware(thunk, epicMiddleware))
```
Comment (amcgee) [inline src/store/index.js]: Is this functionally the same?

File: src/components/app/App.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
+import React, { Component, Fragment } from 'react';
 import PropTypes from 'prop-types';
 import { MuiThemeProvider } from '@material-ui/core/styles';
-import OldMuiThemeProvider from 'material-ui/styles/MuiThemeProvider'; // TODO: REMOVE (currently needed for FileMenu in d2-ui)
+import i18n from '@dhis2/d2-i18n';
+import UI from 'ui/core/UI';
+import HeaderBar from 'ui/widgets/HeaderBar';
 import { muiTheme } from '../../constants/dhis2.theme';
 import MapProvider from '../map/MapProvider';
 import AppHeader from './AppHeader';
```
Comment (amcgee) [inline src/components/app/App.js]: This line should be removed

Comment (turban) [reply src/components/app/App.js]: Done!

Comment (amcgee) [reply src/components/app/App.js]: 👍

File: src/store/index.js (hunk 1 above)
Comment (turban) [reply src/store/index.js]: Not sure - the previous setup gave an error on the store. New syntax is from here:
https://redux-observable.js.org/docs/basics/SettingUpTheMiddleware.html#redux-devtools

Comment (turban) [reply src/store/index.js]: Added a check if in development mode.

**Comment by adeelshahid** [comment]
@turban once this [PR 122](https://github.com/dhis2/ui/pull/122) goes in, z-index styling should be resolved in the HeaderBar

**Comment by adeelshahid** [comment]
@amcgee @turban z-index issues fixed with later PR into dhis2/ui, run `yarn upgrade ui@github:d2-ci/ui`
PR looks to merge

**Pull request merged**